### PR TITLE
Solved issue #7931

### PR DESCRIPTION
**Pull request by Abanoub-Asaad**
Highlight default selection in pickers

## Purpose / Description
This pull request is a solution to the issue num #7931 

## Fixes
Issue #7931 

## Approach
Before change: When the user was in note editor, there is a selector to change the type of note or card's deck, It was not highlited. <br>
After change: Now, when the user selects some thing in this section, it's highlited by default to "gray" color.

## How Has This Been Tested?
Device: Xiaomi redmi note 4

normal
select_type
select_deck

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 1]
```diff
@@ -1397,27 +1429,27 @@ else if (fieldEditText.getText() != null) {
...
-                             * card type was deleted or if the note
-                             * type was changed without moving this
-                             * card to another type. */
-                            Timber.d("onActivityResult() template edit return - current card is gone, close note editor");
-                            UIUtils.showThemedToast(this, getString(R.string.template_for_current_card_deleted), false);
-                            closeNoteEditor();
-                        } else {
-                            Timber.d("onActivityResult() template edit return, in add mode, just re-display");
-                        }
+                // Model can change regardless of exit type - update ourselves and CardBrowser
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:1432]: I do thank you for cleaning the code.
However, may I ask you please to do it in two separate commits. This way, one commit just remove space/add new lines and is really simple to review, and I know where I should concentrate the time I spent reviewing.

Please don't hesitate to ask if you don't know how to split a commit in two

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 2]
```diff
@@ -592,7 +609,22 @@ protected void onCollectionLoaded(Collection col) {
             mAllDeckIds.add(thisDid);
         }
 
-        ArrayAdapter<String> noteDeckAdapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_item, deckNames);
+        ArrayAdapter<String> noteDeckAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_spinner_item, deckNames){
+            @Override
+            public View getDropDownView(int position, View convertView, ViewGroup parent){
```
Comment (Arthur-Milchior) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:614]: Unless you see a real reason not to, could you please factorize this code. That is, use the same code in both functions you add, since they are very similar.

**Review by Arthur-Milchior — COMMENTED** [review]
First, thanks a lot for this code. It's a feature I asked for, that I sometime feel the need to, so that's really cool.

Can you please change the commit name to something more descriptive. This way, if someone look at history, they'll see what is the purpose of this commit in a second and won't loose time on it if they are looking for something else/will find it easily if they need to understand your code.
Something such as

> Current deck has a different colour
>
> Fixes #7931 .

Did you use Android accessibility tool? I'm not an expert in UX, but I fear this gray may be hard to read for some people. There is a tool which automate that check and give best practice advice. In my opinion, either changing the background, or using a darker gray would already help by having a great contrast.

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java (hunk 1 above)
Comment (Abanoub-Asaad) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:1432]: Thank you for your words!
Yes, I tested the feature that I've added by interacting with the app "Manual Testing". <br>
Do you mean making a commit that only has spaces instead of the code that I've written 
and the second commit I'll replace these empty lines with the feature code? 

Another requirement: Should I make another two pull requests, each one has a commit?

Comment (Abanoub-Asaad) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]: Thank you for your words!

You told me to make make 2 commits
So, Can you tell me what each commit should contain?

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]: I'll try to rephrase it, please let me know if it's more clear.

First commit would clean the code. In the example quoted here, it would mean that you'd remove the spaces in order to get correct indentation. That's exactly what is done right here.

In the second commit, you do actual code change, where there is more than change of spaces

Comment (Abanoub-Asaad) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]: Yes, It's clear now
Thank you.

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java (hunk 2 above)
Comment (Abanoub-Asaad) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]: When I used the same code, It showed me an error 
because `new ArrayAdapter<>` isn't allowed in inner classes, so I should make it `new ArrayAdapter<String>` 
and then write out the rest of the implementation.

**Comment by Abanoub-Asaad** [comment]
I updated this PR and made a new one 
Link: https://github.com/ankidroid/Anki-Android/pull/8174

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java (hunk 2 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]: I'd be curious to see the exact error message and what you tried if you please. I cna't imagine why creating an object would be forbidden in an inner class, never saw that

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java]: @Arthur-Milchior 
```
new ArrayAdapter<String> {
     @Override
}
```

Actually creates an anonymous inner class inheriting from `ArrayAdapter<String>`. The extracted code would need to be defined as a class, rather than copy-pasting the variable. I believe this can be done in the "Refactor" context menu of Android Studio, but don't have an IDE on me to check right now.
